source/AI.cpp

```diff
@@ -1019,7 +1058,10 @@ void AI::AskForHelp(Ship &ship, bool &isStranded, const Ship *flagship)
 			if((helper->IsYours() && !ship.IsYours()) || helper.get() == flagship)
 				continue;
 			// Your escorts should not help each other if already under orders.
-			if(helper->IsYours() && ship.IsYours() && orders.count(helper.get()))
+			int helperOrders = 0;
+			if(orders.count(helper.get()))
+				helperOrders = orders.find(helper.get())->second.type;
+			if(helper->IsYours() && ship.IsYours() && orders.count(helper.get()) && helperOrders != Orders::MINING && helperOrders != Orders::HARVEST)
```

Comment:
I'm not sure I understand.  Can you please elaborate?

Reply:
Rather than check `map::count` and then use `map::operator[]`, use `map::find/find_if`. Then you compare against the `.end()` iterator to determine if the order exists, and just dereference the iterator to replace the use of `operator[]`.

Otherwise you iterate the map twice.

Reply:
Fixed by ea0cdf9a30401dda988b0958cd171a4697f28afb